File: app/src/main/java/torille/fi/lurkforreddit/comments/CommentContract.java
package torille.fi.lurkforreddit.comments;

import java.util.List;

import torille.fi.lurkforreddit.BasePresenter;
import torille.fi.lurkforreddit.BaseView;
import torille.fi.lurkforreddit.data.models.view.Comment;
import torille.fi.lurkforreddit.data.models.view.Post;

/**
 * Created by eva on 2/13/17.
 */

public interface CommentContract {

    interface View extends BaseView {

        void showComments(List<Object> comments);

        void showProgressbarAt(int position, int level);

        void hideProgressbarAt(int position);

        void addCommentsAt(List<Comment> comments, int position);

        void showError(String errorText);

        void showErrorAt(int position);

        void setProgressIndicator(boolean active);
    }

    interface Presenter<T extends BaseView> extends BasePresenter<CommentContract.View> {

        void loadComments(String permaLinkUrl, boolean isSingleCommentThread);

        void loadMoreCommentsAt(Comment parentComment, String linkId, int position);
    }
}

File: app/src/main/java/torille/fi/lurkforreddit/comments/CommentContract.kt
package torille.fi.lurkforreddit.comments

import torille.fi.lurkforreddit.BasePresenter
import torille.fi.lurkforreddit.BaseView
import torille.fi.lurkforreddit.data.models.view.Comment

interface CommentContract {

    interface View : BaseView<Presenter> {

        fun showComments(comments: List<Any>)

        fun showProgressbarAt(position: Int, level: Int)

        fun hideProgressbarAt(position: Int)

        fun addCommentsAt(comments: List<Comment>, position: Int)

        fun showError(errorText: String)

        fun showErrorAt(position: Int)

        fun setProgressIndicator(active: Boolean)
    }

    interface Presenter : BasePresenter<View> {

        fun loadComments(permaLinkUrl: String, isSingleCommentThread: Boolean)

        fun loadMoreCommentsAt(parentComment: Comment, linkId: String, position: Int)
    }
}
